website/source/docs/providers/vsphere/r/virtual_machine.html.markdown

```diff
@@ -113,6 +113,7 @@ The `disk` block supports:
 * `template` - (Required if size and bootable_vmdk_path not provided) Template for this disk.
 * `datastore` - (Optional) Datastore for this disk
 * `size` - (Required if template and bootable_vmdks_path not provided) Size of this disk (in GB).
+* `name` - (Required if size is provided) Name of the new disk file to create.
```

Comment:
We could come up with a random string.  I would think providing a name would give good readability in the vSphere environment when the time comes to clean out old disks.  If we auto generate a name, we would have to do checks to ensure it is unique among other (randomly named) disks on that vm.  Would have to read down each disk each time before creating a new disk to ensure no duplicates, or possibly store it in a calculated attribute in the disk schema.